### PR TITLE
Introduction to Python's Virtual Environments and Dependency Managers

### LINKED ISSUE #1887: Introduction to Python's Virtual Environments and Dependency Managers
### Introduction to Python's Virtual Environments and Dependency Managers

When crossing the bridge from being a beginner to an intermediate, you'll start to work on projects. A big problem you'll often encounter is dependency conflicts which differ from one project to another. This is where Python's virtual environments come in.
Virtual environments are isolated Python environments where libraries, python interpreters, and scripts are installed separately from other virtual environments, and the default Python environment installed in the machine.

### Key takeaways: 
1. What are virtual environments?
2. Types of virtual environments in Python
3. How and where to use virtual environments

### References: 
[Python Virtual Environment](https://docs.python.org/3/library/venv.html#:~:text=A%20virtual%20environment%20is%20a,part%20of%20your%20operating%20system.)

### PR DESCRIPTION
**Pull request by FREDERICO23**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1887

### CODE REVIEW

**Comment by geoffrey45** [comment]
@Manley12 This PR is a duplicate of another one that I was working on... which I had requested changes.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@FREDERICO23 Have a look at the comments below. Please work on the suggestions and push the changes.

File: articles/introduction-to- virtual- environments-and-dependency-managers/index.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+```bash
+$ conda deactivate
+```
+
+### 5. Poetry
+**[Poetry](https://python-poetry.org/docs/)** is popularly known as a very feature-rich Python tool for project dependency management. It comes wrapped with a powerful CLI for creating and managing Python projects.
+
+#### Features
+* Rich with features.
```
[the remark this answers is not in the document]
Comment (FREDERICO23) [reply articles/introduction-to- virtual- environments-and-dependency-managers/index.md]: Fixed the requested changes

File: articles/introduction-to- virtual- environments-and-dependency-managers/index.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+### 4. Conda
+**[Conda](https://conda.io/projects/conda/en/latest/index.html)** is a package management and environment management system that not only supports Python but also other languages like Ruby, Scala, R and C/C++. It is used to create, save, load and switch between environments in your local machine.
+
+Conda comes pre-installed in [anaconda](https://www.anaconda.com/) and [miniconda](https://docs.conda.io/en/latest/miniconda.html).  
+
+#### Features
+* You can [share](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#sharing-an-environment) environments.
+* It is rich with features.
+* Finding and installing packages -  There's no need of switching between different environments. Since Conda is also a environment manager, you can set up a another environment to run a different version of Python, without interfering with the environment that runs your usual version of Python.
+
```
[the remark this answers is not in the document]
Comment (FREDERICO23) [reply articles/introduction-to- virtual- environments-and-dependency-managers/index.md]: Done

**Comment by FREDERICO23** [comment]
Made a few more changes for easy understanding by the reader.

**Review by geoffrey45 — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
just need to preview and published - when repo changes are finalized

NOTE: Thank you - we are making some changes to the main repo now - and this is delaying a few article from being published as we want to implement the changes 1st - once these are done - we will publish this one asap. Thank you 🚀

**Comment by FREDERICO23** [comment]
@hectorkambow @geoffrey45 

I did some stuff that broke this pull request. I had opened a new one #2364

**Comment by geoffrey45** [comment]
@FREDERICO23 Your article was published yesterday. Here's the link https://www.section.io/engineering-education/introduction-to-virtual-environments-and-dependency-managers/

**Comment by hectorkambow** [comment]
FYI please be sure to do a git pull on your forked repo before submitting your next PR to avoid any potential merge issues.

Thank you v much again for your patience as we worked the new repo file structure.

**Pull request closed without merging**